src/box/alter.cc

```diff
@@ -3340,9 +3346,11 @@ func_def_new_from_tuple(struct tuple *tuple)
 		}
 		def->param_count = argc;
 		const char *opts = tuple_field(tuple, BOX_FUNC_FIELD_OPTS);
-		if (opts_decode(&def->opts, func_opts_reg, &opts,
-				ER_WRONG_SPACE_OPTIONS, NULL) != 0)
+		if (opts_decode(&def->opts, func_opts_reg, &opts, NULL) != 0) {
+			diag_set(ClientError, ER_WRONG_SPACE_OPTIONS,
```

Comment:
Maybe it should be `ER_WRONG_FUNCTION_OPTIONS`?

Reply:
Yep, it definitely shouldn't be `ER_WRONG_SPACE_OPTIONS`, but I don't want to change this in the scope of this refactoring. Also, this code isn't even covered... I filed a separate ticket to address this issue later, see #7972.